src/hotspot/share/gc/g1/g1FullCollector.hpp

```diff
@@ -87,6 +90,10 @@ class G1FullCollector : StackObj {
   uint                     workers() { return _num_workers; }
   G1FullGCMarker*          marker(uint id) { return _markers[id]; }
   G1FullGCCompactionPoint* compaction_point(uint id) { return _compaction_points[id]; }
+  GrowableArray<HeapRegion*>* skipping_compaction_set(uint id) { return _skipping_compaction_sets[id]; }
+  size_t live_bytes_after_full_gc_mark(uint region_idx) {
+    return MarkSweepDeadRatio > 0 ? _live_stats[region_idx]._live_words * HeapWordSize : 0;
```

Comment:
Not sure if the additional `MarkSweepDeadRatio > 0` check is necessary or useful at all. In case `MarkSweepDeadRatio` is zero, `_hr_live_bytes_threshold` is larger than whatever is stored in here anyway.

Also, please drop the `* HeapWordSize` and compare to a `_hr_live_word_threshold`. This multiplication is completely unnecessary from what I can tell. (It does not really hurt either, but why would I want to multiply both sides of your comparison by the same value before the comparison?)

Also just `liveness()` (or name this and the equivalent in `G1ConcurrentMark` something like `live_words`) is sufficient.